src/frontend/lib/MessagePortLike.ts

```diff
@@ -0,0 +1,119 @@
+import {EventSubscription} from 'react-native';
+import EventEmitter from 'react-native/Libraries/vendor/emitter/EventEmitter.js';
+import {Channel} from 'nodejs-mobile-react-native';
+
+type Listener = (...args: any[]) => void;
+
+export type ServerState = 'idle' | 'started' | 'closed';
+
+export class MessagePortLike extends EventEmitter {
+  #API_EVENT_NAME = '@@API_MESSAGE';
+  #channel: Channel;
+  #eventsSubscriptions = new Map<
+    string,
+    WeakMap<Listener, EventSubscription>
+  >();
+  #channelSubscription: EventSubscription;
+  #state: ServerState = 'idle';
+  #queuedMessages: any[] = [];
+
+  constructor(channel: Channel) {
```

Comment:
Since this is strictly tied to the static import `nodejs-mobile-react-native`.Channel, I think this should just be an import in this file, and not a constructor parameter, so this file encapsulates the Channel dependency.